src/imitation/algorithms/adversarial.py

```
@@ -47,42 +48,49 @@ def __init__(self,
         n_disc_samples_per_buffer: The number of obs-act-obs triples
-            sampled from each replay buffer (expert and generator) during each
-            step of discriminator training. This is also the number of triples
-            stored in the replay buffer after each epoch of generator training.
+          sampled from each replay buffer (expert and generator) during each
+          step of discriminator training. This is also the number of triples
+          stored in the replay buffer after each epoch of generator training.
         gen_replay_buffer_capacity: The capacity of the
-            generator replay buffer (the number of obs-action-obs samples from
-            the generator that can be stored).
+          generator replay buffer (the number of obs-action-obs samples from
+          the generator that can be stored).
 
-            By default this is equal to `20 * n_disc_samples_per_buffer`.
+          By default this is equal to `20 * n_disc_samples_per_buffer`.
         init_tensorboard: If True, makes various discriminator
-            TensorBoard summaries.
+          TensorBoard summaries.
         init_tensorboard_graph: If both this and `init_tensorboard` are True,
-            then write a Tensorboard graph summary to disk.
+          then write a Tensorboard graph summary to disk.
         debug_use_ground_truth: If True, use the ground truth reward for
-            `self.train_env`.
-            This disables the reward wrapping that would normally replace
-            the environment reward with the learned reward. This is useful for
-            sanity checking that the policy training is functional.
+          `self.train_env`.
+          This disables the reward wrapping that would normally replace
+          the environment reward with the learned reward. This is useful for
+          sanity checking that the policy training is functional.
+        use_custom_log: If True, then automatically enter
```

When do we not want to do this?

---

The custom logger as of yet errors if `imitation.util.logger.configure(...)` isn't called ahead of time. Don't want `train()` to error by default because of logging.

Another thing we could do instead of keeping this `use_custom_log` flag is to check inside AdversarialTrainer whether the custom logger is configured, and only enter the accumulate_mean contexts when custom logger is enabled.

We could also consider making `init_trainer` automatically configure the custom logger. Actually I think this is the best solution, as `init_trainer` already takes in `log_dir` and configures `Monitor` logs via `make_vec_env`.

---

Thoughts on the second solution?

---

Second solution sounds good to me.

We should make sure `AdversarialTrainer` fails fast if the user doesn't have the right logger, e.g. add an assert statement somewhere. Don't want to only find this out during training.

---

Ok, moved util.logger.configure into `init_trainer` and out of the `train_adversarial` Sacred script. Also added a fail fast as the first statement of AdversarialTrainer.__init__.